site/docs/hosting/fly.md

```diff
@@ -0,0 +1,426 @@
+# Hosting: Fly <Badge text="Deno" /><Badge text="Node.js" type="warning"/>
+
+This guide tells you about the ways you can host your grammY bots on [Fly](https://fly.io), either using Deno or Node.js.
+
+## Preparing Your Code
+
+You can run your bot either on [webhooks](../guide/deployment-types.md) or [long polling](../guide/deployment-types.md).
+
+### Webhooks
+
+> Remember that you should not call `bot.start()` in your code when using webhooks.
+
+You can use the example code provided in our [Deno Deploy](./deno-deploy.md#preparing-your-code) (Deno) and [Heroku](./heroku.md#express-and-its-middleware) (Node.js) hosting guide.
```

Comment:
This comment should be removed. Imagine someone wants to use Fly, do they now also have to learn what Deno Deploy and Heroku are? Each tutorial must be standalone. If code ends up being identical on two pages, so be it.

Reply:
My point in that sentence was that people could reuse their code (either from Deno deploy or Heroku) in Fly. But, on second thought I think you have a point.